src/mscorlib/shared/System/Text/UTF8Encoding.cs

```diff
@@ -2496,7 +2498,8 @@ public override byte[] GetPreamble()
             else
                 return Array.Empty<byte>();
         }
-
+        
+        public override ReadOnlySpan<byte> Preamble => _emitUTF8Identifier ? s_preamble : Array.Empty<byte>();
 
         public override bool Equals(Object value)
```

Comment:
We may want to get a tracking issue to implement the Span-based overloads on UTF8EncodingSealed that avoids layers of the virtual calls and argument checks.